### DISCUSSION TITLE
Best practices - nested q-layout? q-header?

### DISCUSSION THREAD

**Discussion by saniko** [quasarframework/quasar#12643] [General - Components / Directives / etc] [answered]
Hello,

I'm working with V1.

What is the best practice for creating complex layouts (desktop)?
Can I use nested q-layouts to construct complex layouts? or is q-layout and q-header are singletons?
What if I have a nested layout inside a layout I need to compose, can I do it with the layout components? if so what are the best practices?

Thanks

**Comment by MilosPaunovic** [comment, marked as answer]
It's not great idea to use nested layout, you can check out `QPageContainer` and `QPage` as an alternative.
[https://v1.quasar.dev/layout/page#introduction](https://v1.quasar.dev/layout/page#introduction)
